model.py

```diff
@@ -155,9 +155,23 @@ class Library(Base):
     # The official name of the library.
     name = Column(Unicode, index=True)
 
+    aliases = relationship("LibraryAlias", backref='library')
     service_areas = relationship('ServiceArea', backref='library')
 
     @classmethod
+    def for_name(cls, _db, name):
+        """Find a library whose name or alias matches the given name."""
+
+        # We allow for minor misspellings in the official name,
+        # but not in aliases (which are likely to be acronyms)
+        name_close_enough = func.levenshtein(func.lower(Library.name),
+                                             func.lower(name)) < 2
+        qu = _db.query(Library).outerjoin(Library.aliases).filter(
+            or_(name_close_enough, LibraryAlias.name.ilike(name))
+        )
```

Comment:
This is true, but I'm reluctant to put a lot of work into this until we know more about how people run searches. It might be worth it to create an Elasticsearch index just for library names.